### PR TITLE
Model audit log - snippet / modeladmin history views

### PR DESCRIPTION
**Pull request by gasman**
Incorporates #7409 (generic listing views) and #7550 (logging for non-page models) - new commits start at "Add get_log_entry_models method to log registry".

### CODE REVIEW

File: wagtail/core/models/audit_log.py [hunk 1]
```diff
@@ -80,6 +80,12 @@ def get_for_model(self, model):
     def get_for_user(self, user_id):
         return self.filter(user=user_id)
 
+    def for_instance(self, instance):
+        """
+        Return a queryset of log entries from this log model that relate to the given object instance
+        """
+        raise NotImplementedError  # most be implemented by subclass
```
Comment (jacobtoppm) [inline wagtail/core/models/audit_log.py:87]: most -> must

File: wagtail/snippets/templates/wagtailsnippets/snippets/edit.html [hunk 2]
```diff
@@ -5,6 +5,12 @@
     {% trans "Editing" as editing_str %}
     {% include "wagtailadmin/shared/header_with_locale_selector.html" with title=editing_str subtitle=instance icon="snippet" tabbed=1 merged=1 %}
 
+    {% if latest_log_entry %}
+        <span class="avatar small" data-wagtail-tooltip="{{ latest_log_entry.user_display_name }}"><img src="{% avatar_url latest_log_entry.user size=25 %}" alt="" /></span>
```
Comment (jacobtoppm) [inline wagtail/snippets/templates/wagtailsnippets/snippets/edit.html]: Unfortunately running low on time to try this manually - will do so later - but does this function without the data-wagtail-tooltip js seen on eg https://github.com/wagtail/wagtail/blob/45dce6ac1e41d13878cc655395757d9ffcba6464/wagtail/admin/templates/wagtailadmin/pages/edit.html?

File: wagtail/snippets/templates/wagtailsnippets/snippets/edit.html [hunk 3]
```diff
@@ -9,6 +9,7 @@
         <span class="avatar small" data-wagtail-tooltip="{{ latest_log_entry.user_display_name }}"><img src="{% avatar_url latest_log_entry.user size=25 %}" alt="" /></span>
         {% trans "Last updated" %}
         {% include "wagtailadmin/shared/last_updated.html" with last_updated=latest_log_entry.timestamp time_prefix="at" %}
+        - <a href="{% url 'wagtailsnippets:history' model_opts.app_label model_opts.model_name instance.pk|admin_urlquote %}">History</a>
```
Comment (jacobtoppm) [inline wagtail/snippets/templates/wagtailsnippets/snippets/edit.html]: Shouldn't "History" be translatable here?

File: wagtail/contrib/modeladmin/templates/modeladmin/includes/header_with_history.html [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+<div class="row last-updated">
+    {% if latest_log_entry %}
+    <ul>
+        <li>
+            <span class="avatar small" data-wagtail-tooltip="{{ latest_log_entry.user_display_name }}"><img src="{% avatar_url latest_log_entry.user size=25 %}" alt="" /></span>
+            {% trans "Last updated" %}
+            {% include "wagtailadmin/shared/last_updated.html" with last_updated=latest_log_entry.timestamp time_prefix="at" %}
+        </li>
+        {% if history_url %}
+            <li><a href="{{ history_url }}">{% icon "history" class_name="default" %} History</a></li>
```
Comment (jacobtoppm) [inline wagtail/contrib/modeladmin/templates/modeladmin/includes/header_with_history.html:14]: Needs to be translatable?

File: wagtail/snippets/templates/wagtailsnippets/snippets/_header_with_history.html [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    {% if latest_log_entry %}
+    <div class="row last-updated">
+        <ul>
+            <li>
+                <span class="avatar small" data-wagtail-tooltip="{{ latest_log_entry.user_display_name }}"><img src="{% avatar_url latest_log_entry.user size=25 %}" alt="" /></span>
+                {% trans "Last updated" %}
+                {% include "wagtailadmin/shared/last_updated.html" with last_updated=latest_log_entry.timestamp time_prefix="at" %}
+            </li>
+            {% if history_url %}
+                <li><a href="{{ history_url }}">{% icon "history" class_name="default" %} History</a></li>
```
Comment (jacobtoppm) [inline wagtail/snippets/templates/wagtailsnippets/snippets/_header_with_history.html:16]: Needs to be translatable?

File: wagtail/contrib/modeladmin/templates/modeladmin/create.html [hunk 6]
```diff
@@ -21,6 +21,19 @@
 
     {{ view.media.js }}
     {% prepopulated_slugs %}
+
+    <script>
+        $(function() {
+            $('[data-wagtail-tooltip]').tooltip({
+                animation: false,
+                title: function() {
+                    return $(this).attr('data-wagtail-tooltip');
```
Comment (jacobtoppm) [inline wagtail/contrib/modeladmin/templates/modeladmin/create.html:30]: I do wonder whether we should rely be relying on js to fill in tooltip attributes for accessibility - possibly we should have them there in the DOM (and not in a data-attribute) before the js loads as well. But I think that's a discussion for another time!

**Review by jacobtoppm — APPROVED** [review]
Looks good to me, just missing some translatable strings.

**Comment by gasman** [comment]
@jacobtoppm Good catch on the translations! Rebased with fixes applied now.

**Comment by gasman** [comment]
Merged in 9aa2f68dfa04d5ab21eab059fc5432e7fcfee883

**Pull request closed without merging**